### ISSUE TITLE
Fix external link styling

### ISSUE THREAD

**Issue by mal-tee** [datenanfragen/website#611]
Seems like external links are broken atm, the icon is missing:

![image](https://user-images.githubusercontent.com/45886910/120799457-0f958200-c53f-11eb-9bb8-5cfc3a88569a.png)

**Comment by rugk** [comment]
Also reproducable on https://www.datenanfragen.de/suggest/#!type=new&for=cdb

![grafik](https://user-images.githubusercontent.com/11966684/125464991-ceb4f697-9651-463e-b217-009371836b9a.png)

## Solution

The issue seems to be that strange background-image/size, as it works this way:
![grafik](https://user-images.githubusercontent.com/11966684/125465123-d662d3cb-fa81-4898-9672-847c16fbe15e.png)

**Comment by baltpeter** [comment]
@rugk The external links are supposed to have an icon like this:

![image](https://user-images.githubusercontent.com/4048809/125473900-3acabafc-9b96-4d73-a653-6bdb57926f02.png)

The icon is applied in the `<a>`'s `:after`:

https://github.com/datenanfragen/website/blob/c57db775f69bb13b9a21e00eb438a08ed9e666ea/assets/styles/framework.scss#L214-L219

We then set the `background-size` so that the icon isn't underlined.

---

From what I can tell, the problem is in the SCSS -> CSS compilation.

The correct compiled selector for those rules is (note how the `:after` is at the very end):

```css
main a:not(.button):not(.no-link-decoration):not(.icon):not([href^="https://www.datenanfragen.de"]):not([href^="https://www.datarequests.org"]):not([href^="https://www.demandetesdonnees.fr"]):not([href^="https://www.pedidodedados.org"]):not([href^="https://www.solicituddedatos.es"]):not([href^="https://www.osobnipodaci.org"]):not([href^="https://static.dacdn.de"]):not([href^="http://localhost:"]):not([href^="http://datenanfragen.test:"]):not([href^="https://verein.datenanfragen.de"]):not([href^="#"]):not([href*="--datenanfragen.netlify.app/"]):not([href^="mailto:"]):not([href^="/"])[href*="."]:after {
```

That's also how it was compiled previously. However, recently we're instead seeing this in the production builds:

```css
main a[href*="."]:not([href^="https://static.dacdn.de"]):after:not([href^="/"]):not([href^="mailto:"]):not([href*="--datenanfragen.netlify.app/"]):not([href^="#"]):not([href^="https://verein.datenanfragen.de"]):not([href^="http://datenanfragen.test:"]):not([href^="http://localhost:"]):not([href^="https://www.osobnipodaci.org"]):not([href^="https://www.solicituddedatos.es"]):not([href^="https://www.pedidodedados.org"]):not([href^="https://www.demandetesdonnees.fr"]):not([href^="https://www.datarequests.org"]):not([href^="https://www.datenanfragen.de"]):not(.icon):not(.no-link-decoration):not(.button) {
```

This doesn't match the external links anymore and thus the underline is shorter but no icon is applied.

Unfortunately, so far I haven't managed to reproduce this at all locally, which makes diagnosing the problem further a lot harder than it has to be. :/

**Comment by baltpeter** [comment]
Progress! This seems to be related to the Hugo version. Locally, I was using Hugo 0.85.0 but in CI, we're using 0.83.1. With that, I can reproduce the problem. Let's see if a Hugo upgrade fixes the problem.

**Comment by baltpeter** [comment]
Yep, that seems to work: https://deploy-preview-628--datenanfragen.netlify.app/blog/datenloeschung-bei-werbetreibenden/

#628

**Comment by baltpeter** [comment]
Fixed by #628.
